### ISSUE TITLE
lapack/netlib: Dlantr test suddenly fails on master

### ISSUE THREAD

**Issue by vladimir-ch** [gonum/netlib#68]
Recently Dlantr test started to fail again on master:
```
    dlantr.go:137: norm=I,uplo=U,diag=N,m=2,n=5,lda=5: unexpected result; got 2.435339754899317, want 3.4353397548993168
    dlantr.go:137: norm=I,uplo=U,diag=N,m=2,n=10,lda=10: unexpected result; got 6.8962476950598015, want 7.896247695059803
    dlantr.go:137: norm=I,uplo=U,diag=N,m=3,n=5,lda=5: unexpected result; got 1.9631817230235244, want 2.8302732144124496
    dlantr.go:137: norm=I,uplo=U,diag=N,m=3,n=10,lda=10: unexpected result; got 5.167011030416932, want 6.1670110304169325
    dlantr.go:137: norm=I,uplo=U,diag=N,m=4,n=5,lda=5: unexpected result; got 2.1200546627902757, want 3.0503567512831555
    dlantr.go:137: norm=I,uplo=U,diag=N,m=4,n=10,lda=10: unexpected result; got 7.514745198287458, want 8.514745198287459
```
All failing cases are: inf norm, upper triangular matrix with non-unit diagonal, m < n, n=5 or 10, and lda=n. However, not all test cases which fit these conditions fail. As is usual, I cannot reproduce it locally.

**Comment by kortschak** [comment]
The only obvious difference is the sha of OpenBLAS. I have bumped up the frequency of the cron  builds to weekly so that we can catch regressions there more easily.

The last successful cron build used xianyi/OpenBLAS@eb2eddf241dd5e46e0064cd311b6a63229897d21 and the most recent (failing) build used xianyi/OpenBLAS@f95989cbc1d0c8dedbded1183103569cd64584fd.

I'll try making the OpenBLAS version use the last good commit and see if that fixes the failure.

**Comment by kortschak** [comment]
... and of course, that last known good version [also fails](https://travis-ci.org/gonum/netlib/builds/617412068).

**Comment by vladimir-ch** [comment]
I can reproduce the failures locally if I replace the import path in go.mod to the local gonum checkout (the modules are still confusing me). As before, the result is actually correct, I just don't see why the test would compute an incorrect expected value and only in certain cases. I again suspect some memory corruption in LAPACKE. To that end, I compiled OpenBLAS with my recent commits in the reference lapacke but then running the test crashes with "double free or corruption" ᕙ(⇀‸↼‶)ᕗ I'll continue looking into this as my patience permits.

**Comment by kortschak** [comment]
What commit ref does your replace point to to get the replication? and what OpenBLAS were you using? Why commit ref for OpenBLAS causes this new error?

**Comment by vladimir-ch** [comment]
gonum and OpenBLAS at latest commits, only with `lapacke_dlantr_work.c` taken from latest reference lapack. I think the issue will be in the transpose function `LAPACKE_dtr_trans`. I also think it is actually not necessary, it should be possible to call `LAPACK_dlantr` directly with `a` and appropriately adapted arguments. I'll do some experiments tomorrow.

**Comment by vladimir-ch** [comment]
I think I'm onto something. I'll confirm tomorrow, it's too late here.

**Comment by vladimir-ch** [comment]
This was "fixed" in #69 by using v0.6.2 of the main gonum module which does not test matrix sizes that may cause memory corruption in Fortran DLANTR.
